stone/target/python_rsrc/stone_base.py

```diff
@@ -34,6 +34,16 @@ def __init__(self, tag, value=None):
         self._tag = tag
         self._value = value
 
+    def __eq__(self, other):
+        # Also need to check if one class is a subclass of another. If one union extends another,
+        # the common fields should be able to be compared to each other.
+        return (
+            (isinstance(self, other.__class__) or isinstance(other, self.__class__)) and
```

Comment:
Yes. See discussion at https://tails.corp.dropbox.com/D227666:

union A
   t0

union B extends A
   t1

A(t0) should be equal to B(t0).